### DISCUSSION TITLE
How to detect collisions

### DISCUSSION THREAD

**Discussion by emortaldev** [Minestom/Minestom#1519] [Help]
I want to detect an enderpearl hitting a block. I currently check if any of the velocity axis are 0, however if an enderpearl is shot directly upwards it "collides" instantly

**Comment by GoldenGamerLP** [comment]
Look for gravity ticks, if they are zero it landed (edit: I did it like this)

**Reply by emortaldev** [reply]
What if it hits a wall?

**Reply by GoldenGamerLP** [reply]
The gravity tick is then zero

**Reply by emortaldev** [reply]
this doesn't work for colliding with walls

**Reply by GoldenGamerLP** [reply]
Then check if any of the x, y, z axes of the velocity is zero

**Reply by emortaldev** [reply]
> I currently check if any of the velocity axis are 0

That is where the issue is

**Reply by GoldenGamerLP** [reply]
Have you opened a issue then?
